cocos/2d/CCLabel.cpp

```diff
@@ -447,6 +446,7 @@ Label::~Label()
 
     CC_SAFE_RELEASE_NULL(_textSprite);
     CC_SAFE_RELEASE_NULL(_shadowNode);
+    CC_SAFE_RELEASE_NULL(_reusedLetter);
```

Comment:
what is the reason? if really need I think we should move ```        Node::removeAllChildrenWithCleanup(true);
        CC_SAFE_RELEASE_NULL(_reusedLetter);``` both line, just like the method `Label::reset()` does

Reply:
The problem seems to come from line 605, where `_reusedLetter` is manually retained and is never released if `_fontAtlas` is nullptr when the Label is deleted.

Calling `removeAllChildrenWithCleanup(true);` might be unnecessary as the profiler (Instuments.app) told that there was no leak after the modification.

Reply:
when I test using cpp-empty-tests, I find at method `setFontAtlas()` member:`_fontAtlas` and `_reusedLetter` init together, and at method `reset()` set nullptr together, in fact I check this program using _instruments->leaks_, didn't find memory leaks. I didn't find the state: __fontAtlas is nullptr, \_reusedLetter not nullptr__` can you describe how you used, It's better to show some code.

Reply:
Here are screenshots of Instruments:
![a_beforepatch](https://user-images.githubusercontent.com/422538/33423027-f12e1bee-d5b7-11e7-89fb-e5c50a209449.jpg)
![b_afterpatch](https://user-images.githubusercontent.com/422538/33423028-f143dd08-d5b7-11e7-9ced-fc8b06f8c0f1.jpg)

Reply:
You are right, leaks occur in this method

The state __fontAtlas is nullptr, \_reusedLetter not nullptr__ appear, when I make a condition breakpoint `(_fontAtlas==nullptr)&&(_reusedLetter!=nullptr)` and run `57:System Font Test` of cpp-tests. 
After some analysis, I found it cased by `Label::updateContent`, in this method `_fontAtlas` is clear, `_reusedLetter` not.

and the test code is
```
    auto label = Label::createWithTTF("This is a very long sentence", "fonts/arial.ttf", 20);
    label->setSystemFontSize(40); 
    label->setPosition(Vec2(visibleSize / 2) + origin);
    this->addChild(label, 1);
```
only `createWithTTF` leaks doesn't occur

Reply:
so we also can add `CC_SAFE_RELEASE_NULL(_reusedLetter)` to `Label::updateContent()`, keep the both `_fontAtla`  and `_reusedLetter` clear at same time. This leaks will be fixed also.
```void Label::updateContent()
{
    ...
    CC_SAFE_RELEASE_NULL(_reusedLetter);
    CC_SAFE_RELEASE_NULL(_textSprite);
    CC_SAFE_RELEASE_NULL(_shadowNode);
    ...
}
```
will you test this leaks using this method? I think it will also works. @UsualYann

Reply:
Ok this worked, I've pushed the change. Thanks :)